File: Xtra-Kick(Alpha).apk_Decompiler.com/sources/com/github/andreyasadchy/xtra/model/chat/Emote.java
package com.github.andreyasadchy.xtra.model.chat;

import android.support.v4.media.h;
import xa.j;

public abstract class Emote {
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        return (obj instanceof Emote) && j.a(getName(), ((Emote) obj).getName());
    }

    public abstract String getName();

    public String getType() {
        return null;
    }

    public abstract String getUrl1x();

    public abstract String getUrl2x();

    public abstract String getUrl3x();

    public abstract String getUrl4x();

    public int hashCode() {
        return getName().hashCode();
    }

    public Boolean isAnimated() {
        return null;
    }

    public boolean isZeroWidth() {
        return false;
    }

    public String toString() {
        return h.m(Constants.EXT_TAG_END, getName());
    }
}

File: app/src/main/java/com/github/andreyasadchy/xtra/model/chat/Emote.kt
package com.github.andreyasadchy.xtra.model.chat

class Emote(
    val name: String? = null,
    val localData: Pair<Long, Int>? = null,
    val url1x: String? = null,
    val url2x: String? = null,
    val url3x: String? = null,
    val url4x: String? = null,
    val format: String? = null,
    val isAnimated: Boolean = true,
    val isOverlayEmote: Boolean = false,
    val source: Int? = null,
) {
    val thirdParty = source == PERSONAL_STV || source == CHANNEL_STV || source == GLOBAL_STV

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Emote || name != other.name) return false
        return true
    }

    override fun hashCode(): Int {
        return name.hashCode()
    }

    override fun toString(): String {
        return ":$name"
    }

    companion object {
        const val PERSONAL_STV = 0
        const val CHANNEL_STV = 1
        const val GLOBAL_STV = 2
    }
}
